socs/agents/camera_rtsp/agent.py

```diff
@@ -0,0 +1,887 @@
+        )
+        cnts = imutils.grab_contours(cnts)
+
+        # loop over the contours
+        img_area = thresholded.shape[0] * thresholded.shape[1]
+        min_area = self.min_frac * img_area
+        max_area = self.max_frac * img_area
+        detection = False
+        for c in cnts:
+            # if the contour is too small or too large, ignore it
+            area = cv2.contourArea(c)
+            if area < min_area:
+                continue
+            if area > max_area:
+                continue
+            # compute the bounding box for the contour, draw it on the frame,
+            # and update the text
+            detection = True
+            (x, y, w, h) = cv2.boundingRect(c)
+            cv2.rectangle(data, (x, y), (x + w, y + h), (0, 255, 0), 2)
+        return (data, detection)
+
+
+class FakeCamera:
+    """Class used generate image data on demand for testing.
+
+    Args:
+        width (int):  Width of each frame in pixels.
+        height (int):  Height of each frame in pixels.
+        fps (float):  The frames per second of the stream.
+
+    """
+
+    def __init__(self, width=1280, height=720, fps=20.0):
+        self.width = width
+        self.height = height
+        self.fps = fps
+
+        # The number of frames to hold fixed before switching
+        self.fixed_frames = int(60 * self.fps)
+
+        # The size of a square to randomly place in the field of view
+        self.square_dim = 100
+        self.sq_half = self.square_dim // 2
+        self._random_sq_pos()
+
+        self.current = None
+        self.frame_count = 0
+
+    def isOpened(self):
+        return True
+
+    def _random_sq_pos(self):
+        sq_y = int(self.height * np.random.random_sample(size=1)[0])
+        sq_x = int(self.width * np.random.random_sample(size=1)[0])
+        if sq_y > self.height - self.sq_half:
+            sq_y = self.height - self.sq_half
+        if sq_y < self.sq_half:
+            sq_y = self.sq_half
+        if sq_x > self.width - self.sq_half:
+            sq_x = self.width - self.sq_half
+        if sq_x < self.sq_half:
+            sq_x = self.sq_half
+        self.sq_x = sq_x
+        self.sq_y = sq_y
+
+    def grab(self):
+        img = np.zeros((self.height, self.width, 3), dtype=np.uint8)
+        img[:, :, 0] = 127
+        img[:, :, 1] = 127
+        img[:, :, 2] = 127
+        if self.frame_count % self.fixed_frames == 0:
+            self._random_sq_pos()
+        img[
+            self.sq_y - self.sq_half: self.sq_y + self.sq_half,
+            self.sq_x - self.sq_half: self.sq_x + self.sq_half,
+            0,
+        ] = 255
+        self.current = img
+        self.frame_count += 1
+        return True
+
+    def retrieve(self):
+        img = np.array(self.current)
+        self.current = None
+        return True, img
+
+    def read(self):
+        _ = self.grab()
+        return self.retrieve()
+
+    def release(self):
+        pass
+
+
+class CameraRTSPAgent:
```

Comment:
Nitpicky, but could you update this and all other instances to `RTSPCameraAgent` for consistency with the `ACTiCameraAgent`?

Reply:
Sure, I was aiming for `CategorySpecificThingAgent`, but looking across the codebase now I see most agents are `SpecificThingCategoryAgent`.  I'll change it.

Reply:
This is implemented.